php/src/Docker/DockerActionManager.php

```diff
@@ -425,10 +409,18 @@ public function CreateContainer(Container $container) : void {
             $requestBody['HostConfig']['SecurityOpt'] = ["apparmor:unconfined"];
         }
 
+        $mounts = [];
+        foreach($container->GetVolumes()->GetVolumes() as $volume) {
+            if (!str_starts_with($volume->name, '/')) {
+                $mounts[] = ["Type" => "volume", "Source" => $volume->name, "Target" => $volume->mountPoint, "ReadOnly" => !$volume->isWritable];
+            } else {
+                $mounts[] = ["Type" => "bind", "Source" => $volume->name, "Target" => $volume->mountPoint, "ReadOnly" => !$volume->isWritable, "Propagation" => "rshared"];
```

Comment:
This means it will bug out if a directory is not existing irrc. need to test this.

Also test if this works on docker desktop

Reply:
indeed this is not going to work.

Reply:
@chatziko we have a problem here. using mounts instead of volumes will bug out if you specify a directory as target that is not existing yet which we definitely don't want to happen. So we can either have this behaviour kept of not bugging out in that case or setting the propagation to rshared. Not sure how to solve this. WDYT?

Reply:
So from what I understand (I did a few tests myself), the mount directory inside the container is always created. But if the one in the host does not exist:
- `--mount type=bind,source=/foo,target=/foo` will fail with an error
- `--volume /foo:/foo` will create an empty dir `/foo` in the host

Here's an idea: if `--mount` fails, catch the error and re-run the container with `--volume`.
- This will be completely transparent to the user.
- It will only happen the first time, since on future restarts the directory will exist (it gets created).

Another option would be to add a new env variable that controls the propagation, and explain that it can only be set if the mount dir exists.

Reply:
> if `--mount` fails, catch the error and re-run the container with `--volume`.

I fear this is not really possible as checking for that specific error is not going to work....

Actually I am also not sure about an additional config value for this. Will need to think about a good solution to this.